src/client/components/pages/Courses/CourseInstanceTable.tsx

```diff
@@ -67,6 +80,62 @@ const CourseInstanceTable: FunctionComponent<CourseInstanceTableProps> = ({
       viewColumn === COURSE_TABLE_COLUMN.ENROLLMENT
     ));
 
+  /**
+   * The current value for the metadata context
+   */
+  const metadata = useContext(MetadataContext);
+
+  /**
+   * The current value of the area filter dropdown
+   */
+  const [areaValue, setAreaValue] = useState<string>('All');
+
+  /**
+   * The current value of the "Is SEAS" filter dropdown
+   */
+  const [isSEASValue, setIsSEASValue] = useState<string>('All');
+
+  /**
+   * The current value of the fall semester "offered" filter dropdown
+   */
+  const [fallOfferedValue, setFallOfferedValue] = useState<string>('All');
+
+  /**
+   * The current value of the spring semester "offered" filter dropdown
+   */
+  const [springOfferedValue, setSpringOfferedValue] = useState<string>('All');
+
+  const filteredCourses = (currentCourses:
+  { course: CourseInstanceResponseDTO, element: ReactElement }[]):
+  { course: CourseInstanceResponseDTO, element: ReactElement }[] => {
+    let courses = currentCourses;
+    if (areaValue !== 'All') {
+      courses = listFilter(
+        courses,
+        { field: 'course.area', value: areaValue, exact: true }
+      );
+    }
+    if (isSEASValue !== 'All') {
+      courses = listFilter(
+        courses,
+        { field: 'course.isSEAS', value: isSEASValue, exact: true }
+      );
+    }
+    if (fallOfferedValue !== 'All') {
+      courses = listFilter(
+        courses,
+        { field: 'course.fall.offered', value: fallOfferedValue, exact: true }
+      );
+    }
+    if (springOfferedValue !== 'All') {
+      courses = listFilter(
+        courses,
+        { field: 'course.spring.offered', value: springOfferedValue, exact: true }
+      );
+    }
+    return courses;
+  };
+
```

Comment:
I think all of this filtering logic needs to be moved up a level, into the `CoursesPage` component. The idea is that we want the `CourseInstanceTable` to just be concerned with how the table itself gets rendered, while the parent component can handle the fetching and manipulation of the data.

Doing that would mean that we'd need to pass down handlers for filtering the different fields. So I think we should also simplify the structure of our state by consolidating all the filter fields into a `filters` state, e.g.

```ts

interface SemesterFilterState {
  offered: OFFERED | 'all';
  // and so on
}

interface FilterState {
  area: string;
  isSEAS: IS_SEAS | 'all';
  spring: SemesterFilterState;
  fall: SemesterFilterState;
  // and so on
}

const [filters, setFilters] = useState<FilterState>({
  area: 'all',
  isSEAS: 'all',
  spring: {
    offered: 'all',
  },
  fall: {
    offered: 'all'
  },
});
```

Then we can use the [functional update][1] argument to `setFilters` to create a generic handler that can be passed down. The basic look of this would be something like this, though you'd need to account for the nested semester field: 

```jsx
const genericFilterUpdate = (field, value) => {
  setFilters((filters) => ({
    ...filters,
    [field]: value,
  }));
};

/* In our Dropdown component */
<Dropdown
  id="area"
  onChange={(evt) => { genericFilterUpdate('area', evt.target.value); }}
/>
<Dropdown
  id="isSEAS"
  onChange={(evt) => { genericFilterUpdate('isSEAS', evt.target.value); }}
/>
```

(That might be tricky to type out the function signature in Typescript, you'd probably need to use some combination of [generics][2], [overloads][3], and [`keyof` types][4])

[1]: https://reactjs.org/docs/hooks-reference.html#functional-updates
[2]: https://www.typescriptlang.org/docs/handbook/2/functions.html#generic-functions
[3]: https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads
[4]: https://www.typescriptlang.org/docs/handbook/2/keyof-types.html

Reply:
Thanks for your help with this! The filtering logic has been moved to the `CoursesPage` level now so that we can pass down the `filteredCourses` to the `CourseInstanceTable.`